controllers/imagecollector/imagecollector_controller.go

```diff
@@ -222,6 +274,139 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{RequeueAfter: *repeatPeriod}, nil
 }
 
+func (r *Reconciler) getChildImageJobs(ctx context.Context, collector *eraserv1alpha1.ImageCollector) ([]eraserv1alpha1.ImageJob, error) {
+	imageJobList := &eraserv1alpha1.ImageJobList{}
+	if err := r.List(ctx, imageJobList); err != nil {
+		log.Info("could not list imagejobs")
+		return nil, err
+	}
+
+	relevantJobs := util.FilterJobListByOwner(
+		imageJobList.Items, metav1.NewControllerRef(collector, schema.GroupVersionKind{
+			Group:   "eraser.sh",
+			Version: "v1alpha1",
+			Kind:    "ImageCollector",
+		}),
+	)
+
+	return relevantJobs, nil
+}
+
+func (r *Reconciler) upsertImageList(ctx context.Context, collector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	imageListItems := make([]string, 0, len(collector.Status.Vulnerable))
+	images := collector.Status.Vulnerable
+
+	if *deleteScanFailedImages {
+		images = append(images, collector.Status.Failed...)
```

Comment:
why does this take in more than one argument?

Reply:
`append` requires the first argument to be a slice `[]T` and the remaining arguments to all be of type `T`

Reply:
sorry I meant the ... following status failed

Reply:
That just passes in the failed images one by one. If there are `n` images, this is the same as:
```
images = append(images, collector.Status.Failed[0], collector.Status.Failed[1], ..., collector.Status.Failed[n-1])
```

Reply:
got it thanks!